### PR TITLE
refactoring: filters/aggregations new implementation

### LINKED ISSUE #74: Filters/Aggregations: fix nested and nice urls
- [ ] refactor filters to support correctly nested filters
- [ ] allow filter selection via url args with nice urls (currently it is encoding something like `[filter=value]`

### PR DESCRIPTION
**Pull request by ntarocco**
- creates new components for buckets aggregation and filters
- re-implements logic around filters
- renames some configuration keys to improve clarity
- allows more configuration of URL params
- changes sort components to be displayed even if 1 result
- adds tests
- adds some documentation
- improves demos
- closes #74 #57 #6 #7

### CODE REVIEW

File: src/lib/components/BucketAggregation/BucketAggregation.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+BucketAggregation.propTypes = {
+  title: PropTypes.string.isRequired,
+  agg: PropTypes.shape({
+    field: PropTypes.string.isRequired,
+    aggName: PropTypes.string.isRequired,
+    childAgg: PropTypes.object,
+  }),
+  userSelectionFilters: PropTypes.array.isRequired,
```
Comment (zzacharo) [inline src/lib/components/BucketAggregation/BucketAggregation.js]: In general I think we should find an elegant way of distinguishing the props that this component accepts and the props coming from redux..Just for a developer to understand from the propTypes which one this component accepts.

File: docs/docs/using_url_parameters.md [hunk 2]
```diff
@@ -90,19 +91,28 @@ class MyUrlParser {
...
 ```
 
+### Disable deep linking
+
+Deep linking can be disabled by setting to `false` the `persistentUrl.enabled` config variable.
+
 ### Inject the new configuration
 
-You can inject overridden configuration in an object named `urlParamsConfig`.
+You can inject overridden configuration in an object named `persistentUrl`.
```
[the remark this answers is not in the document]
Comment (ntarocco) [reply docs/docs/using_url_parameters.md]: changed!

File: docs/docs/using_url_parameters.md [hunk 3]
```diff
@@ -90,19 +91,28 @@ class MyUrlParser {
...
 const myUrlParser = new MyUrlParser();
 
 <ReactSearchKit
-  urlParamsConfig={{
-    urlParamsMapping: myUrlParamsMapping,
-    paramValidator: myParamValidator,
-    urlParser: myUrlParser,
+  searchApi={...}
+  persistentUrl={{
+    enabled: true,
```
[the remark this answers is not in the document]
Comment (ntarocco) [reply docs/docs/using_url_parameters.md]: done!

File: src/lib/components/BucketAggregation/BucketAggregation.js (hunk 1 above)
Comment (ntarocco) [reply src/lib/components/BucketAggregation/BucketAggregation.js]: https://github.com/inveniosoftware/react-searchkit/issues/82

**Pull request merged**